src/main/java/reposense/authorship/analyzer/AnnotatorAnalyzer.java

```diff
@@ -22,12 +22,18 @@
  */
 public class AnnotatorAnalyzer {
     private static final String AUTHOR_TAG = "@@author";
-    // reference: https://github.com/shinnn/github-username-regex
+    // GitHub username format
     private static final String REGEX_AUTHOR_NAME_FORMAT = "^[a-zA-Z0-9](?:[a-zA-Z0-9]|-(?=[a-zA-Z0-9])){0,38}$";
     private static final Pattern PATTERN_AUTHOR_NAME_FORMAT = Pattern.compile(REGEX_AUTHOR_NAME_FORMAT);
-    private static final String REGEX_COMMENT_WITH_AUTHOR_TAG_FORMAT = "^[^\\w]*@@author(\\s[^\\s]+)?[\\s]*$";
-    private static final Pattern PATTERN_COMMENT_WITH_AUTHOR_TAG_FORMAT =
-            Pattern.compile(REGEX_COMMENT_WITH_AUTHOR_TAG_FORMAT);
+    private static final String REGEX_AUTHOR_TAG_FORMAT = "@@author(\\s[^\\s]+)?";
+
+    private static final String[][] COMMENT_FORMATS = {
+            {"//", "\\s"},
```

Comment:
Actually, can you change this part to define these patterns directly? Otherwise, everytime `checkValidCommentLine` is called, the same pattern may need to be complied again

Reply:
Done